src/escrow_contract/src/main.nr

```diff
@@ -31,20 +30,21 @@ pub contract Escrow {
         owner: PrivateImmutable<AddressNote, Context>,
     }
 
-    // Creates a new instance
+    // Creates a new instance of the escrow contract
     #[private]
     #[initializer]
     fn constructor(owner: AztecAddress) {
-        let mut note = AddressNote::new(owner, owner);
-        storage.owner.initialize(&mut note).emit(encode_and_encrypt_note_unconstrained(
-            &mut context,
-            // NOTE: assuming non-factory deployment (msg_sender = Alice)
-            // TODO: encrypt for Alice
-            owner, // to: Bob
-            context.this_address(), // from: Escrow
-        ));
+        let sender = context.msg_sender();
+        let mut note = AddressNote::new(owner, sender);
+        let note_emission = storage.owner.initialize(&mut note);
+        note_emission.emit(encode_and_encrypt_note(&mut context, owner, context.this_address()));
+        note_emission.emit(encode_and_encrypt_note(&mut context, sender, context.this_address()));
```

Comment:
in the case of the clawback escrow, we're emitting this note_emission for the clawback contract (a contract w/o keys), so i don't think is needed to do this. 

anyone who has the escrow keys should be able to fetch who the owner is:

```suggestion
        let self_address = context.this_address();
        note_emission.emit(encode_and_encrypt_note(&mut context, self_address, self_address));
```